src/renderer/components/node/Node.tsx

```diff
@@ -147,6 +155,68 @@ const NodeInner = memo(({ data, selected }: NodeProps) => {
     const bgColor = useColorModeValue('gray.400', 'gray.750');
     const shadowColor = useColorModeValue('gray.600', 'gray.900');
 
+    const inputDataValues = useMemo(() => Object.values(inputData), [inputData]);
+
+    const [shouldRun, setShouldRun] = useState(false);
+    useEffect(() => {
+        setShouldRun(
+            validity.isValid &&
+                !inputs.some((i) => i.hasHandle) &&
+                inputDataValues.length >= inputs.filter((i) => !i.optional).length
```

Comment:
This determines if a node should make a run/individual request. The conditions are:
1) it has no inputs that have handles (meaning they don't require input from other nodes, we can just send all the inputData we have)
2) the input data we have covers all non-optional inputs

I'm not really sure why I did that second one, to be honest. It might have just been something I was trying when I had an unrelated bug and I just never removed it. But theoretically it wouldn't be necessary to keep that since the validity checker already checks this, right?

Reply:
Ah, I know what this was for now: When you clear a node, it detects that an input data change happened before it recomputes validity, and therefore it tries to run it again and errors